activerecord/test/cases/secure_token_test.rb

```diff
@@ -22,4 +22,11 @@ def test_regenerating_the_secure_token
     assert_not_equal @user.token, old_token
     assert_not_equal @user.auth_token, old_auth_token
   end
+
+  def test_token_value_not_overwritten_when_present
```

Comment:
The callback used to generate a token is triggered before creating a record. Your test uses an existing record and the callback won't be triggered.